wd_helper.au3

```diff
@@ -636,6 +636,199 @@ Func _WD_FrameLeave($sSession)
+	Local $a_Result[0][6], $sStartLocation = '', $sMessage = ''
+	Local $vResult = '', $iErr = $_WD_ERROR_Success, $iExt = 0
+	Local $sStartBody_ElementID = _WD_ExecuteScript($sSession, "return window.document.body;", Default, Default, $_WD_JSON_Element)
+
+	If $sFilter <> '' Then $bReturnHTML = True
+	If @error Then
+		$iErr = $_WD_ERROR_GeneralError
+	Else
+		$vResult = __WD_FrameList_Internal($sSession, 'null', '', $bReturnHTML)
+		$iErr = @error
+	EndIf
+	#Region - _WD_FrameList - post processing
+	If $iErr = $_WD_ERROR_Success Then ; can occur when $sFilter is used
+
+		; Strip last @CRLF
+		$vResult = StringTrimRight($vResult, 2)
+
+		; create array of frames from string returned from __WD_FrameList_Internal
+		_ArrayAdd($a_Result, $vResult)
+
+		; find "calling frame" - set $sStartLocation
+		For $i = 0 To UBound($a_Result) - 1
+			If $a_Result[$i][4] = $sStartBody_ElementID Then $sStartLocation = $a_Result[$i][0]
+		Next
+
+		; recalculate absolute path COL0 to relative path COL1
+		For $i = 0 To UBound($a_Result) - 1
+			$a_Result[$i][1] = StringRegExpReplace($a_Result[$i][0], '\A' & $sStartLocation & '\/?', '')
+		Next
+
+		; apply filter
+		If $sFilter <> '' Then
+			For $i = UBound($a_Result) - 1 To 0 Step -1
+				; check $s_HTML content :  $a_Result[$i][5]
+				If StringRegExp(BinaryToString($a_Result[$i][5], $SB_UTF8), $sFilter, $STR_REGEXPMATCH) = 1 Then
+					; do nothing keeping current row of $a_Result[$i][....]
+				ElseIf @error = 2 Then ; @error StringRegExp : Bad pattern. @extended = offset of error in pattern
+					$iExt = @extended
+					$iErr = $_WD_ERROR_InvalidExpression
+					ExitLoop
+				Else ; StringRegExp = 0 no MATCH
+					_ArrayDelete($a_Result, $i)
+				EndIf
+			Next
+			If UBound($a_Result) = 0 Then
+				$iErr = $_WD_ERROR_NotFound
+			EndIf
+		EndIf
+
+		; select desired DataType for the $vResult
+		If $bReturnAsArray Or $bReturnHTML Then ; do not return huge amount of HTML data to string - only as array as they can contain restricted data
+			$vResult = $a_Result
+		Else ; string is option for testing and asking support thus should not return huge amount of HTML data to string as they can contain restricted data
+			$vResult = _ArrayToString($a_Result)
+		EndIf
+	EndIf
+
+	; Back to "calling frame"
+	If $sStartLocation Then
+		_WD_FrameEnter($sSession, $sStartLocation)
+		$iErr = @error
+	EndIf
+
+	If $iErr And $iErr <> $_WD_ERROR_InvalidExpression Then
+		$sMessage = 'Was not able to check / back to "calling frame"'
+	EndIf
+
+	#EndRegion - _WD_FrameList - post processing
+
+	$sMessage = ($sMessage And $_WD_DEBUG = $_WD_DEBUG_Full) ? (' Information: ' & $sMessage) : ("")
+	Return SetError(__WD_Error($sFuncName, $iErr, $sParameters & $sMessage), $iExt, $vResult)
+EndFunc   ;==>_WD_FrameList
+
+; #INTERNAL_USE_ONLY# ===========================================================================================================
+; Name ..........: __WD_FrameList_Internal
```

Comment:
@Danp2 please reaview `header` and in code `comments` for `__WD_FrameList_Internal`